xDrip/Scenes/Settings/SettingsUserType/SettingsPenUser/SettingsPenUserViewController.swift

```diff
@@ -49,23 +49,223 @@ class SettingsPenUserViewController: BaseSettingsViewController, SettingsPenUser
     }
     
     // MARK: IB
+    private lazy var tableView: UITableView = {
+        var tableView = UITableView(frame: .zero, style: tableViewStyle)
+        
+        tableView.keyboardDismissMode = .onDrag
+        
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        tableView.delegate = self
+        tableView.dataSource = self
+        view.addSubview(tableView)
+        tableView.bindToSuperview()
+        
+        tableView.registerNib(type: BaseSettingsDisclosureCell.self)
+        tableView.registerNib(type: PickerExpandableTableViewCell.self)
+        
+        return tableView
+    }()
+    
+    private var tableViewStyle: UITableView.Style {
+        if #available(iOS 13.0, *) {
+            return .insetGrouped
+        } else {
+            return .grouped
+        }
+    }
+    
+    private var addBarButton: UIBarButtonItem?
+    private var viewModel: BaseSettings.ViewModel?
+    internal var expandedCell: IndexPath?
     
     // MARK: View lifecycle
     
     override func viewDidLoad() {
         super.viewDidLoad()
+        createAddButton()
         doLoad()
     }
     
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        showAddButton()
+    }
+    
     // MARK: Do something
     
     private func doLoad() {
-        let request = SettingsPenUser.Load.Request()
-        interactor?.doLoad(request: request)
+        let request = SettingsPenUser.UpdateData.Request(animated: false)
+        interactor?.doUpdateData(request: request)
+    }
+    
+    private func createAddButton() {
+        let barButton = UIBarButtonItem(barButtonSystemItem: .add, target: self, action: #selector(onAdd))
+        addBarButton = barButton
+    }
+    
+    private func showAddButton() {
+        parent?.parent?.navigationItem.rightBarButtonItem = addBarButton
     }
     
     // MARK: Display
     
-    func displayLoad(viewModel: SettingsPenUser.Load.ViewModel) {
+    func displayUpdateData(viewModel: SettingsPenUser.UpdateData.ViewModel) {
+        update(with: viewModel.tableViewModel, animated: viewModel.animated)
+    }
+    
+    // MARK: Handlers
+    
+    @objc private func onAdd() {
+        let request = SettingsPenUser.Add.Request()
+        interactor?.doAdd(request: request)
+    }
+    
+    private func update(with tableViewModel: BaseSettings.ViewModel, animated: Bool) {
+        self.viewModel = tableViewModel
+        if animated {
+            UIView.transition(
+                with: tableView,
+                duration: 0.35,
```

Comment:
Let's define duration as a global constant. Not sure about this case, but usually animation duration is `0.25`